src/ui/public/i18n/context.tsx

```diff
@@ -16,23 +16,32 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+
 import React from 'react';
+import { I18nStartContract } from '../../../core/public/i18n';
+
+export let I18nContext: I18nStartContract['Context'] = null!;
+export function __newPlatformInit__(context: typeof I18nContext) {
+  if (I18nContext) {
+    throw new Error('ui/i18n/context already initialized with new platform apis');
+  }
 
-import { I18nProvider } from './provider';
+  I18nContext = context;
+}
 
-export function injectI18nProvider<P>(WrappedComponent: React.ComponentType<P>) {
-  const I18nProviderWrapper: React.SFC<P> = props => {
+export function wrapIntoI18nContext<P>(ComponentToWrap: React.ComponentType<P>) {
```

Comment:
__note:__ this is basically workaround for `reactDirective` and I didn't want to have it in the core, hence declaring it here.

Reply:
`wrapInto...` is pretty unusual phrasing, can we use `wrapIn...` or `wrapWith...`?

Reply:
Haha, silly me :) Will change, thanks!